aggregator/src/main/java/com/redhat/cloud/notifications/db/EmailAggregationRepository.java

```diff
@@ -21,12 +24,40 @@ public class EmailAggregationRepository {
     @Inject
     EntityManager entityManager;
 
+    @Inject
+    AggregatorConfig aggregatorConfig;
+
     /**
      * Get applications with pending aggregation it the current time matches with org aggregation time preference.
      * Results will be limited to last 48h
      */
     public List<AggregationCommand> getApplicationsWithPendingAggregationAccordingOrgPref(LocalDateTime now) {
         LocalDateTime currentTimeTwoDaysAgo = now.minusDays(2);
+        List<Object[]> records;
+        if (aggregatorConfig.isFetchAggregationsWithAtLeastOneSubscriber()) {
+            final LocalDateTime startTime = LocalDateTime.now();
+            String query = "SELECT DISTINCT ev.orgId, ev.bundleId, ev.applicationId, acp.lastRun, bu.name, ap.name FROM Event ev " +
+                "join Application ap on ev.applicationId = ap.id join Bundle bu on ev.bundleId = bu.id " +
+                "join AggregationOrgConfig acp on ev.orgId = acp.orgId WHERE " +
+                // check than at least one user of the org subscribed for daily digest with this event type
+                "EXISTS (SELECT 1 FROM EventTypeEmailSubscription es where ev.orgId = es.id.orgId and es.id.subscriptionType='DAILY' and es.eventType = ev.eventType and es.subscribed is true) " +
+                // check for linked email integration linked to this event type (to honor legacy mechanism)
+                "AND EXISTS (SELECT 1 FROM Endpoint ep, EndpointEventType eet where (ev.orgId = ep.orgId or ep.orgId is null) and ep.compositeType.type = 'EMAIL_SUBSCRIPTION' and eet.eventType = ev.eventType and eet.endpoint = ep) " +
```

Comment:
Asking for clarification: in what scenarios do endpoints not have an org ID? I see the existing query in this method does the same, I'm just wondering about the underlying reason.

Reply:
ah right, thanks for explaining!